### PR TITLE
both king and knight valid moves, and tests, on top of master of Jan1…

### PR DESCRIPTION
**Pull request by falimo**
0 team session.  this PR replaces the other commits/PRs related to the valid-move for both king and knight.

### CODE REVIEW

File: app/models/knight.rb [hunk 1]
```diff
@@ -1,4 +1,54 @@
...
+  elsif (self.x_position - move_to_x) === 1 && (self.y_position - move_to_y) === 2
+    return true
+  elsif (self.x_position - move_to_x) === -1 && (self.y_position - move_to_y) === 2
+    return true
+  elsif (self.x_position - move_to_x) === 2 && (self.y_position - move_to_y) === 1
+    return true
+  elsif (self.x_position - move_to_x) === -2 && (self.y_position - move_to_y) === -1
+    return true
+  elsif (self.x_position - move_to_x) === -1 && (self.y_position - move_to_y) === -2
+    return true
```
Comment (sinkoffr) [inline app/models/knight.rb:27]: Farouk,  I think lines 12 through 26 can be condensed down if you use the built in ruby ".abs" method like you have in the king model.

**Comment by sinkoffr** [comment]
I think this is looking good.  There are a few of the houndci-bot comments that should be corrected and I left an additional comment on the knight model that will decrease the lines of code and make it look more cohesive with the king model (as well as the rook and queen methods I have written.

File: app/models/knight.rb (hunk 1 above)
Comment (falimo) [reply app/models/knight.rb:27]: great suggestion - thanks!  have made the changes - much simpler - and will resubmit.

**Comment by falimo** [comment]
have made changes (to use .abs method) suggested by rebecca. will resubmit with these changes and clean-up of houndci-bot suggestions.

**Pull request closed without merging**